cdap-security/src/main/java/io/cdap/cdap/security/server/AbstractLoginService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.server;
+
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+/** AbstractLoginService
+ * AbstractLoginService to provide support for username case conversion
+ */
+public abstract class AbstractLoginService extends AbstractLifeCycle implements LoginService {
+
+    private static final Logger LOG = Log.getLogger(AbstractLoginService.class);
+    protected CConfiguration cConfiguration;
+    protected String usernameCaseConversion = "";
+    protected char[] usernameCamelCaseConversionDelimiterArr = {'.'};
+
+    protected AbstractLoginService() {
+        cConfiguration = CConfiguration.create();
```

Comment:
Please don't create cConf. It should be passed to constructor. You should hook it up with guice injector.

Reply:
Using this code now
`Injector injector = Guice.createInjector(new ConfigModule());
    this.cConfiguration = injector.getInstance(CConfiguration.class);`

I am not familiar with Guice, please check if it is the right way.
I have tested it, it works.